src/transformers/models/seamless_m4t/modeling_seamless_m4t.py

```diff
@@ -0,0 +1,4058 @@
+            idx_most_probable_sequences_per_batch = idx_most_probable_sequences_per_batch + torch.arange(batch_size)*num_return_sequences
+            t2u_input_embeds = t2u_input_embeds[idx_most_probable_sequences_per_batch]
+            sequences = sequences[idx_most_probable_sequences_per_batch]
+
+        # TODO: is it the proper way, what's the priority with generation config and so on?
+        pad_token_id = self.config.pad_token_id
+
+        # Compute new attention mask
+        seq_lens = (sequences != pad_token_id).int().sum(1)
+        t2u_model_attention_mask = to_attention_mask(t2u_input_embeds, seq_lens)
+        kwargs_speech_generation["attention_mask"] = t2u_model_attention_mask
+
+        output_speech = self.t2u_model.generate(inputs_embeds=t2u_input_embeds, **kwargs_speech_generation)
+        # TODO: proper output form
+
+        return output_speech
+
+    def prepare_inputs_for_generation(
+        self,
+        decoder_input_ids,
+        past_key_values=None,
+        attention_mask=None,
+        head_mask=None,
+        decoder_head_mask=None,
+        cross_attn_head_mask=None,
+        use_cache=None,
+        encoder_outputs=None,
+        **kwargs,
+    ):
+        # cut decoder_input_ids if past is used
+        if past_key_values is not None:
+            decoder_input_ids = decoder_input_ids[:, -1:]
+
+        return {
+            "input_ids": None,  # encoder_outputs is defined. input_ids not needed
+            "encoder_outputs": encoder_outputs,
+            "past_key_values": past_key_values,
+            "decoder_input_ids": decoder_input_ids,
+            "attention_mask": attention_mask,
+            "head_mask": head_mask,
+            "decoder_head_mask": decoder_head_mask,
+            "cross_attn_head_mask": cross_attn_head_mask,
+            "use_cache": use_cache,
+        }
+
+
+############ VOCODER related code ################
+
+
+class SeamlessM4TVariancePredictor(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+
+
+class SeamlessM4TVocoder(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+
+        self.conv_pre = ...  # Conv1d(...)
+
+        self.ups = nn.ModuleList([])  # ... ConvTranspose1d
+        self.resblocks = nn.ModuleList([])  # ... RESBLOCKS
+
+        self.conv_post = ...  # Conv1d(...)
+
+        self.dict_embeds_layer = nn.Embedding(...)  #
+        self.spkr_embeds_layer = nn.Embedding(...)  #
+        self.lang_embeds_layer = nn.Embedding(...)  #
+
+        self.dur_predictor = SeamlessM4TVariancePredictor()
+
+
+# TODO: model with vocoder head
+
+
+############ WHOLE MODEL related code ################
+
+
+@add_start_docstrings(
+    "The bare SeamlessM4T Model transformer outputting raw hidden-states without any specific head on top.",
+    SEAMLESS_M4T_START_DOCSTRING,
+)
+class SeamlessM4TModelOld(SeamlessM4TPreTrainedModel):
```

Comment:
Out of interest - what makes this model "Old?"

Reply:
Is this just a debugging class @ylacombe?